### PR TITLE
Add test for POM DSL using Kotlin DSL

### PR DESCRIPTION
**Pull request by marcphilipp**
Original Issue: #3348

### CODE REVIEW

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+        testDirectory.file('settings.gradle.kts')
+    }
+
+    def "can customize POM using Kotlin DSL"() {
+        given:
+        settingsFile << 'rootProject.name = "customizePom"'
+        buildFile << """
+            plugins {
+                id("maven-publish")
+            }
```
Comment (eskatos) [inline subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy]: :o:
```kotlin
plugins {
    `maven-publish`
}
```

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+            group = "org.gradle.test"
+            version = "1.0"
+
+            publishing {
+                repositories {
+                    maven { url = uri("${mavenRepo.uri}") }
```
Comment (eskatos) [inline subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy]: :o:
```kotlin
repositories {
    maven(url = "${mavenRepo.uri}")
}
```

File: subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                }
+                publications {
+                    val mavenCustom by creating(MavenPublication::class) {
```
Comment (eskatos) [inline subprojects/maven/src/integTest/groovy/org/gradle/api/publish/maven/MavenPublishPomCustomizationKotlinDslIntegTest.groovy]: :o: 
```kotlin
publications {
    create<MavenPublication>("mavenCustom") {
        // ..
    }
}
```

**Comment by marcphilipp** [comment]
@eskatos Thanks for the feedback – applied in the updated commit!

Out of curiosity, how does the syntax with the backticks work? Is there a reference for it somewhere?

```kotlin
plugins {
    `maven-publish`
}
```

**Comment by eskatos** [comment]
Backticks are used to escape member names that are not valid kotlin identifiers, here because of `-`:
```kotlin
plugins {
    application
    `maven-publish`
}
```

**Comment by bamboo** [comment]
https://kotlinlang.org/docs/reference/java-interop.html#escaping-for-java-identifiers-that-are-keywords-in-kotlin

**Comment by JLLeitschuh** [comment]
This is really nice, is this feature currently in `4.7` or will it be available in `4.8`?

**Comment by marcphilipp** [comment]
@eskatos @bamboo Thanks! 🙂 

@JLLeitschuh It will be in 4.8.

**Pull request merged**